sklearn/impute/tests/test_impute.py

```diff
@@ -1061,9 +1069,11 @@ def test_iterative_imputer_skip_non_missing(skip_complete):
         assert_allclose(X_test_est[:, 0], [11, 7, 12], rtol=1e-4)
 
 
-@pytest.mark.parametrize("rs_imputer", [None, 1, np.random.RandomState(seed=1)])
-@pytest.mark.parametrize("rs_estimator", [None, 1, np.random.RandomState(seed=1)])
-def test_iterative_imputer_dont_set_random_state(rs_imputer, rs_estimator):
+@pytest.mark.parametrize("test_none_state_imputer", [True, False])
+@pytest.mark.parametrize("test_none_state_estimator", [True, False])
+def test_iterative_imputer_dont_set_random_state(
+    test_none_state_imputer, test_none_state_estimator, global_random_seed
+):
```

Comment:
I think we should leave this test unchanged. It doesn't seem to use the random state to generate random numbers, instead it only checks if an attribute is set correctly and not modified. I don't think we need to test this with multiple random seeds (via the `global_random_seed` fixture)